### PR TITLE
feat: Go reference docs

### PR DESCRIPTION
**Pull request by HomelessDinosaur**

### CODE REVIEW

File: src/nav.config.ts [hunk 1]
```diff
@@ -1043,6 +1062,247 @@ const fullNav: FullNav = {
...
+        },
+      ],
+    },
+    {
+      title: 'APIs',
+      links: [
+        {
+          title: 'api()',
+          href: '/reference/go/v0/api/api',
+        },
```
Comment (tjholm) [inline src/nav.config.ts:1080]: I wonder if for this section, if we should add a top level `resources` reference?

So we'd have

resources:
 - NewApi() -> Api
 - NewQueue() -> Queue
 - NewTopic() -> Topic
 - ... etc

Followed by the relevent API level calls and (new resources would also link into these

Api
 - Get()
 - Post()
 - Put()
 - Delete()
 - ... etc ...

Comment (HomelessDinosaur) [reply src/nav.config.ts:1080]: Yeah I think that's a better way of organising it. I'll make the change now.

**Review by davemooreuws — CHANGES_REQUESTED** [review]
Looks good. 

Seems like they are some weird indenting issues in the code examples. Such as the one in apis.mdx:43 and 60

**Review by davemooreuws — APPROVED** [review]

File: src/pages/reference/go/v0/api/api-delete.mdx [hunk 2]
```diff
@@ -0,0 +1,197 @@
+export const description =
+  "Reference for Nitric's Go library - Register an API route and set a specific HTTP DELETE handler on that route."
+
+# Go - Api.Delete()
+
+Register an API route and set a specific HTTP DELETE handler on that route.
+
+<Note>
+  This method is a convenient short version of
+  [Api.route.Delete()](./api-route-delete)
```
Comment (tjholm) [inline src/pages/reference/go/v0/api/api-delete.mdx:10]:
```suggestion
  [Api.Route.Delete()](./api-route-delete)
```

Comment (tjholm) [reply src/pages/reference/go/v0/api/api-delete.mdx:10]: Apart from that, LGTM

**Review by jyecusch — APPROVED** [review]

**Review by tjholm — APPROVED** [review]

**Pull request merged**